kolibri/content/management/commands/retrievecontent.py

```diff
@@ -75,27 +74,19 @@ def handle_network_download(self, *args, **options):
                     url = paths.get_content_storage_file_url(filename)
                     path = paths.get_content_storage_file_path(filename)
 
-                    try:
-                        filedir = os.path.dirname(path)
-                        os.makedirs(filedir)
-                    except OSError:  # directories already exist
-                        pass
+                    # if the file already exists, add its size to our overall progress, and skip
+                    # TODO(jamalex): could do md5 checks here instead, to be ultra-safe
+                    if os.path.isfile(path) and os.path.getsize(path) == f.file_size:
+                        overall_progress_update(f.file_size)
+                        continue
 
-                    r = requests.get(url, stream=True)
-                    r.raise_for_status()
-                    contentlength = int(r.headers['content-length'])
+                    with transfer.FileDownload(url, path) as download:
 
-                    with self.start_progress(total=contentlength) as file_dl_progress_update:
+                        with self.start_progress(total=download.total_size) as file_dl_progress_update:
 
-                        with open(path, "wb") as destfileobj:
-
-                            for content in r.iter_content(1000):
-                                length = len(content)
-
-                                destfileobj.write(content)
-
-                                overall_progress_update(length)
-                                file_dl_progress_update(length)
+                            for chunk in download:
```

Comment:
Since this is the length of the chunk returned, not the chunk itself, there should be a better name for this.